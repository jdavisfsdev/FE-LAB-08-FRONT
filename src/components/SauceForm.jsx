/* eslint-disable max-len */
import React, { useState } from 'react';
import SauceFormCSS from './SauceForm.module.css';
import { addSauce } from '../utils/sauceApi';

export default function SauceForm() {
  const [name, setName] = useState('');
  const [url, setUrl] = useState('');
  const [price, setPrice] = useState('');
  const [heat, setHeat] = useState('');

  const handleNameChange = ({ target }) => {
    setName(target.value);
  };

  const handlePriceChange = ({ target }) => {
    setPrice(target.value);
  };

  const handleUrlChange = ({ target }) => {
    setUrl(target.value);
  };

  const handleHeatChange = ({ target }) => {
    setHeat(target.value);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const sauce = {
      name,
      url,
      price,
      heat
    };
    console.log(sauce);
    addSauce(sauce);
    alert('Sauce Added!');
  };

  return (
    <form onSubmit={handleSubmit} className={SauceFormCSS.form}>
      <p>
        <label>
          <span>Sauce Name</span>
          <input type="text" name="name" placeholder="Name" onChange={handleNameChange}/>
        </label>
      </p>

      <p>
        <label>
          <span>Sauce Image URL</span>
          <input type="text" name="url" placeholder="URL" onChange={handleUrlChange}/>
        </label>
      </p>

      <p>
        <label>
          <span>Sauce Price</span>
          <input type="text" name="price" placeholder="Price" onChange={handlePriceChange}/>
        </label>
      </p>

      <p>
        <label>
          <span>Sauce Heat</span>
          <input type="text" name="heat" placeholder="Heat"  onChange={handleHeatChange}/>
        </label>
      </p>

      <button onClick={handleSubmit}>SUBMIT</button>
    </form>
  );
}
