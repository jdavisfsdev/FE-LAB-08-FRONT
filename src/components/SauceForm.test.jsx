import React from 'react';
import { render } from '@testing-library/react';
import SauceForm from './SauceForm';

describe('SauceForm component snapshot', () => {
  it('render Sauce form', () => {
    const { asFragment } = render(
      <SauceForm />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
