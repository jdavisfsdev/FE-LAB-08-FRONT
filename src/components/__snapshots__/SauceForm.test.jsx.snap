// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SauceForm component snapshot render Sauce form 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <p>
      <label>
        <span>
          Sauce Name
        </span>
        <input
          name="name"
          placeholder="Name"
          type="text"
        />
      </label>
    </p>
    <p>
      <label>
        <span>
          Sauce Image URL
        </span>
        <input
          name="url"
          placeholder="URL"
          type="text"
        />
      </label>
    </p>
    <p>
      <label>
        <span>
          Sauce Price
        </span>
        <input
          name="price"
          placeholder="Price"
          type="text"
        />
      </label>
    </p>
    <p>
      <label>
        <span>
          Sauce Heat
        </span>
        <input
          name="heat"
          placeholder="Heat"
          type="text"
        />
      </label>
    </p>
    <button>
      SUBMIT
    </button>
  </form>
</DocumentFragment>
`;
